aspnet/tutorials/web-api-help-pages-using-swagger.rst

```diff
@@ -0,0 +1,328 @@
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Controllers/TodoController.cs
+    :language: c#
+    :start-after: Delete_Method
+    :end-before: #endregion
+    :dedent: 8
+    :emphasize-lines: 2
+
+.. image:: web-api-help-pages-using-swagger/_static/triple-slash-comments.png
+
+Note that the UI is driven by the generated JSON file, and these comments are also in that file as well.
+
+.. code-block:: javascript
+    :emphasize-lines: 5
+
+      "delete": {
+        "tags": [
+          "Todo"
+        ],
+        "summary": "Deletes a specific TodoItem",
+        "operationId": "ApiTodoByIdDelete",
+        "consumes": [],
+        "produces": [],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content"
+          }
+        },
+        "deprecated": false
+      }
+
+Here is a more robust example, adding ``<remarks />`` where the content can be just text or adding the JSON or xml object for further documentation of the method.
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Controllers/TodoController.cs
+    :language: c#
+    :start-after: Create_Method
+    :end-before: #endregion
+    :dedent: 8
+    :emphasize-lines: 4-14
+
+Notice the enhancement of the UI with these additional comments.
+
+.. image:: web-api-help-pages-using-swagger/_static/xml-comments-extended.png
+
+
+DataAnnotations
+'''''''''''''''
+Another supported feature is using the ``System.ComponentModel.DataAnnotations`` to decorate the API controller, methods and classes to help drive the Swagger components.
+
+For example, adding the ``[Required]`` annotation to the ``Name`` property of the TodoItem class will change the ModelSchema information in the UI. Other DataAnnotations such as ``[Produces("application/json"]``, RegularExpression validators and more will further detail the information delivered in the generated page.  What is important is that the more information that is documented in the code, the more "auto" generated information is pushed to the consumer.
+
+
+.. literalinclude:: web-api-help-pages-using-swagger/sample/src/TodoApi/Models/TodoItem.cs
+  :language: c#
+  :emphasize-lines: 10
+
+
+Describing Response Types
+'''''''''''''''''''''''''
+Consuming developers are probably most concerned with what is returned; specifically response types, error codes (if not standard). These are handled in the xml comments and DataAnnotations.
+
+Take the ``Create()`` method for example, currently it returns only "201 Created" response by default. That is of course if the item is in fact created, or a "204 No Content" if no data is passed in the POST Body.  However, there is no documentation to know that or any other response. That can be fixed by adding the following piece of code.
```

Comment:
The ``Create`` method returns 201 when an item is created and 204 if .... You would need to review the code to know the possible returns, it's not available in the Swagger UR. In the following code well (explain to them what you're adding)